### DISCUSSION TITLE
One click to create adhoc filters based on log attributes

### DISCUSSION THREAD

**Discussion by RoryCrispin** [grafana/clickhouse-datasource#402] [Ideas]
The log panel detail field is great, I love that you can see the distribution of logs with each attribute. A common pattern is to take a value from the detail and filter by it. 
For instance, I see that a lot of errors are in the region ‘eu-west-2’, I have to copy the value then
 The flow here is to click copy the field and then create an ad-hoc filter and paste the value. I wish I could click a button here and have it create an ad-hoc filter immediately (no copy and paste)

<img width="416" alt="Screenshot 2023-05-30 at 11 05 50" src="https://github.com/grafana/clickhouse-datasource/assets/716269/84b4bb2d-5380-463a-a43a-f41803eddb81">

This seems to be supported in Grafana already using the eye button (although I think this just filters the current log panel rather than adding an ad-hoc filter) https://grafana.com/static/img/docs/explore/data-link-9-4.png
https://grafana.com/docs/grafana/latest/explore/logs-integration/

**Comment by mshustov** [comment]
> The log panel detail field is great, I love that you can see the distribution of logs with each attribute. A common pattern is to take a value from the detail and filter by it.
> For instance, I see that a lot of errors are in the region ‘eu-west-2’, I have to copy the value then

Unfortunately, Grafana's panel API doesn't seem to support this functionality.
See the Grafana-Elasticsearch playground https://play.grafana.org/d/PXyvC66nz/elasticsearch-8-2-0-sample-web-logs?orgId=1
Instead, it's recommended to use [Template variables](https://grafana.com/docs/grafana/latest/dashboards/variables/add-template-variables/)

> https://grafana.com/docs/grafana/latest/explore/logs-integration/

The link you attached is about ad-hoc filters in `Expore` view only. The Datasource implemented this functionality in v3.1.0. Please, consider upgrading to access this functionality.

**Comment by SpencerTorres** [comment]
Old thread, but this feature is available
![image](https://github.com/grafana/clickhouse-datasource/assets/28887171/64b99e69-8dbc-4508-abd5-bfad98121c00)
